packages/dds/tree/src/test/testNodeId.ts

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { fail } from "assert";
+import { NodeId } from "../feature-libraries/index.js";
+import {
+	ChangeEncodingContext,
+	DeltaFieldMap,
+	FieldKey,
+	FieldKindIdentifier,
+	makeAnonChange,
+} from "../core/index.js";
+import { JsonCompatibleReadOnly, brand } from "../index.js";
+// eslint-disable-next-line import/no-internal-modules
+import { EncodedNodeChangeset } from "../feature-libraries/modular-schema/modularChangeFormat.js";
+import { TestChange } from "./testChange.js";
+
+/**
+ * A NodeId containing a TestChange.
+ * This is intended for the purpose of testing that `FieldChangeRebaser`s correctly handle child nodes.
+ */
+export interface TestNodeId extends NodeId {
+	readonly testChange: TestChange;
+}
+
+export const TestNodeId = {
+	composeChild,
+	rebaseChild,
+	deltaFromChild,
+	create,
+	encode,
+	decode,
+};
+
+function create(id: NodeId, testChange: TestChange): TestNodeId {
+	return { ...id, testChange };
+}
+
+function composeChild(
+	id1: NodeId | undefined,
+	id2: NodeId | undefined,
+	verify: boolean = true,
+): TestNodeId {
+	const testChange = TestChange.compose(tryGetTestChange(id1), tryGetTestChange(id2), verify);
+	const resultId = id1 ?? id2 ?? fail("Should not compose two undefined IDs");
+	const composed: TestNodeId = {
+		...resultId,
+		testChange,
+	};
+
+	return composed;
+}
+
+function rebaseChild(
+	idToRebase: NodeId | undefined,
+	baseId: NodeId | undefined,
+): TestNodeId | undefined {
+	const testChange = TestChange.rebase(tryGetTestChange(idToRebase), tryGetTestChange(baseId));
+	const resultId = idToRebase ?? baseId ?? fail("Should not rebase two undefined IDs");
```

Comment:
Why would we ever want to return `baseId` here?

Reply:
We could get rid of it as it will never actually be used since `TestChange.rebase` will always return undefined in that case. However, in production ModularChangeFamily does use the base ID in this case, as we need a new ID to use to represent this node in case any changes are moved into it as part of the rebase.